### PR TITLE
Add Debian 7 full support for new package promise.

### PR DESCRIPTION
**Pull request by pasinskim**

### CODE REVIEW

**Comment by pasinskim** [comment]
@kacfengine this should fix all deb based distros we are supporting.

File: modules/packages/apt_get [hunk 1]
```diff
@@ -238,8 +244,13 @@ def repo_install():
 
     args = package_arguments_builder(True)
 
-    if args:
-        return subprocess_call(cmd_line + args, stdout=sys.stderr)
+    if (not args):
+        return 0
+
+    a1, a2 =  map(list, zip(*args))
```
Comment (kacfengine) [inline modules/packages/apt_get]: I think this could use a comment to explain what it does.

File: modules/packages/apt_get [hunk 2]
```diff
@@ -198,9 +197,16 @@ def one_package_argument(name, arch, version, is_apt_install):
...
     if archs:
-        args = [name + ':' + arch + version_suffix for arch in archs]
+        for cur_arch in archs:
+            if cur_arch == platform_arch:
+                if is_apt_install:
+                    args.append((name + version_suffix, name + version_suffix))
+                else:
+                    args.append((name + version_suffix, name + ':' + cur_arch + version_suffix))
+            else:
+                args.append((name + ':' + cur_arch + version_suffix, name + ':' + cur_arch + version_suffix))
```
Comment (kacfengine) [inline modules/packages/apt_get]: A comment would be useful here as well, to explain why we add the two different variants.

**Comment by kacfengine** [comment]
Very nice way of solving it though!

**Comment by pasinskim** [comment]
@kacfengine I have added some comments to explain what is happening there...

**Comment by kacfengine** [comment]
Cherry-picked to 3.7.x as well.

**Pull request merged**